### PR TITLE
fix: upgrade openpgp

### PR DESCRIPTION
**Pull request by 0618**
#### Description of changes

Resolve dependabot alert https://github.com/aws-amplify/amplify-cli/security/dependabot/193

Migrate openpgp from v4 to v5 by following the instructions https://github.com/openpgpjs/openpgpjs/releases/tag/v5.0.0

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by awsluja — APPROVED** [review]

File: packages/amplify-util-mock/src/__e2e_v2__/test-synthesizer/transform.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export const defaultTransformParams: Pick<ExecuteTransformConfig, 'transformersF
     enableAutoIndexQueryNames: true,
     respectPrimaryKeyAttributesOnConnectionField: true,
     enableSearchNodeToNodeEncryption: false,
+    enableTransformerCfnOutputs: true,
```
Comment (0618) [inline packages/amplify-util-mock/src/__e2e_v2__/test-synthesizer/transform.ts:20]: This change is to resolve

Screenshot 2023-09-27 at 9 54 56 PM

File: packages/amplify-graphiql-explorer/package.json [hunk 2]
```diff
@@ -29,7 +29,7 @@
     "eslint-webpack-plugin": "^3.1.1",
     "file-loader": "^6.2.0",
     "fs-extra": "^10.0.0",
-    "graphiql": "^1.5.16",
+    "graphiql": "1.5.16",
```
Comment (0618) [inline packages/amplify-graphiql-explorer/package.json:32]: This change is to resolve the following build error

image

With `^`, `graphiql` would use `1.11.x`

File: packages/amplify-util-mock/src/__e2e_v2__/test-synthesizer/transform.ts (hunk 1 above)
Comment (sobolk) [reply packages/amplify-util-mock/src/__e2e_v2__/test-synthesizer/transform.ts:20]: @alharris-at did data ship a break ?

Comment (0618) [reply packages/amplify-util-mock/src/__e2e_v2__/test-synthesizer/transform.ts:20]: The break is from catacory-api v3.2.0 and the code is here https://github.com/aws-amplify/amplify-category-api/commit/0c72d1822f8e5ccb3e04a0a49049a459b5fb49e6

Comment (0618) [reply packages/amplify-util-mock/src/__e2e_v2__/test-synthesizer/transform.ts]: It turned out that I deleted and regenerated `yarn.lock` when merging `dev` eb3c0bcaff330979bfeab730da8b94455d4a0249

File: packages/amplify-graphiql-explorer/package.json (hunk 2 above)
Comment (0618) [reply packages/amplify-graphiql-explorer/package.json]: This change is not necessary anymore since `yarn.lock` is fixed

**Pull request closed without merging**